### ISSUE TITLE
HVG selection and need for reference mapping

### ISSUE THREAD

**Issue by Nusob888** [theislab/expiMap_reproducibility#24]
Hi authors, 

Thank you for expimap, it is a brilliant concept and one I am keen to use. I am not sure if this is the best platform to be asking this and happy to transfer to discourse if appropriate. 

I had 2 questions:
1) How does highly variable gene selection influence the ability to learn new gene programs?
I note based on the notebooks and vignettes, that hvg selection is made on the reference dataset (unperturbed). Then the "perturbed" datasets are reference mapping using the same genes and their annotated programs. 
- Would this not affect the ability to learn independent gene programs that are not directly captured by the initial hvg selection? since some lowly expressed genes may become up-regulated in certain conditions and not be captured from an unperturbed reference dataset.  
- Has this ever been benchmarked?

2) On the same note, is there an need to reference map?
- Could one integrate both query and reference datasets in the same model?
- If so, is there a way to include independent adaptors to learn new unannotated GPs?

Many thanks in advance for your help and insight.

**Comment by Koncopd** [comment]
Hello, @Nusob888, thanks for your interest.

1. Yes, this could affect the learning of independent genes programs in the query. So if you have access to perturbed and unperturbed, it is better to get hvg from the composite data. Also you can take higher number of hvg, for our examples in the paper we had from 2k (pbmc) to 10k (pancreas).
2. You can integrate them together if you have access. However, to learn new GPs, we would advise to do query to reference projection. It is needed, because freezing the GPs from the reference allows to learn new GPs in the query (perturbed), otherwise even with higher independence constraints they are normally too flexible compared to the annotated GPs and tend to take a lot of variation from them.

**Comment by Nusob888** [comment]
Thanks for the insight @Koncopd,

1. It is good to know that I can treat the paired control and condition datasets as a complete dataset to analyse without reference mapping. 
2.  This is good to know. However, would this suffer the same issue as point 1? can gene programs only be learned from those constrained by the reference dataset's hvg selection?

In some ways, running the model on a large number of hvgs e.g. 10K makes sense to capture as much variability as possible in anticipation for learning new programs.

**Comment by Koncopd** [comment]
2. I am not sure i get the point. If you have 2 datasets, you can run HVG on them both, train one as a reference and then do query to reference mapping of the second simultaneously learning new GPs.

**Comment by Nusob888** [comment]
Sorry, I probably wasn't clear. 

One of the main selling points in the manuscript and the documentation is that you can reference map to reference data for atlas'ing and alternate analysis approaches. 

In that scenario, you would envision a repository of pre-trained models of reference data, without having to retrain per analysis right?

If I have two datasets, a reference and a query... sure, I can run hvg on both to maximise capturing the variance across **both** datasets. 

But.. that is not what is done in the manuscript/reproducibility code and is not what is advised in the documentation. In those, hvgs were selected on the reference datasets first. Also, selecting on hvgs on control and perturbed datasets each time would defeat the point/benefit of having a reference dataset/model. If I have to re-select hvgs every time I have a new perturbed dataset, then the benefit of reference mapping goes out of the window. 

I am not suggesting that I can't use the approach you mention re: selecting hvg on both. I was just wondering/opening the discussion on whether or not there are ways around missing "unseen" gene variances when using a pre-trained reference model approach. e.g. should be selecting more hvgs as you suggest, or whether we should just be re-training per analysis.

**Comment by Koncopd** [comment]
Yes, you are right, in the standard setting i would propose to have higher number of hvg in the reference.

It would be interesting to think about alternative approaches to include new additional genes in the query, which do not require retraining with reference included, but i am not sure i have anything in my mind right now.
